### PR TITLE
eject 0.1.12 (new formula)

### PR DESCRIPTION
**Pull request by edwardloveall**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/eject.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  def install
+    xcodebuild
+    bin.install "build/Release/eject.app/Contents/MacOS/eject"
+    frameworks_path = "build/Release/eject.app/Contents/Frameworks"
+    mv(frameworks_path, frameworks)
+  end
+
+  test do
+    system "#{bin}/eject"
```
Comment (edwardloveall) [inline Formula/eject.rb]: I'm aware that this is a pretty weak test. Right now the tool only prints out usage instructions or requires a xib which would be a lot of lines to embed for a test. If you think that's reasonable or there's a better way, let me know and I'll change it.

Comment (MikeMcQuaid) [inline Formula/eject.rb]: What's the smallest possible `xib` that can be created?

File: Formula/eject.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  url "https://github.com/Raizlabs/Eject/archive/0.1.12.tar.gz"
+  sha256 "a4dae3d37f780d274f53ed25d9dc1a27d5245289f9b8cbaaf8be71bc9334de18"
+
+  depends_on :xcode => :build
+
+  def install
+    xcodebuild
+    bin.install "build/Release/eject.app/Contents/MacOS/eject"
+    frameworks_path = "build/Release/eject.app/Contents/Frameworks"
+    mv(frameworks_path, frameworks)
```
Comment (bfontaine) [inline Formula/eject.rb]: You can drop the parentheses here.

**Comment by bfontaine** [comment]
The [build failed](https://bot.brew.sh/job/Homebrew%20Core/11358/version=yosemite/testReport/junit/brew-test-bot/yosemite/install_eject/) as well as the [audit](https://bot.brew.sh/job/Homebrew%20Core/11358/version=any_sierra/testReport/junit/brew-test-bot/sierra/audit_eject___new_formula/):
```
Error: 2 problems in 1 formula
eject:
  * GitHub repository not notable enough (<20 forks, <20 watchers and <50 stars)
  * GitHub repository too new (<30 days old)
```

File: Formula/eject.rb (hunk 1 above)
Comment (edwardloveall) [inline Formula/eject.rb]: Looks like I can create one that's 16 lines long, which doesn't seem that bad:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="11134" systemVersion="15F34" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" colorMatched="YES">
    <dependencies>
        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="11106"/>
        <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
    </dependencies>
    <objects>
        <placeholder placeholderIdentifier="IBFilesOwner" id="-1" userLabel="File's Owner"/>
        <placeholder placeholderIdentifier="IBFirstResponder" id="-2" customClass="UIResponder"/>
        <view contentMode="scaleToFill" id="iN0-l3-epB">
            <rect key="frame" x="0.0" y="0.0" width="375" height="667"/>
            <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
            <color key="backgroundColor" red="1" green="1" blue="1" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
        </view>
    </objects>
</document>
```

When run through eject, it creates the following output:

```swift

// Create Views
let view = UIView()
view.frame = CGRect(x: 0, y: 0, width: 375, height: 667)
view.autoresizingMask = [.flexibleHeight, .flexibleWidth]
view.backgroundColor = UIColor(red: 1, green: 1, blue: 1, alpha: 1)
```

Comment (edwardloveall) [inline Formula/eject.rb]: Ok, I created a better test. Let me know what you think!

**Comment by edwardloveall** [comment]
Oh, I see it's failing on the repository not being popular enough. I guess I'll do a tab instead. Sorry!

**Comment by MikeMcQuaid** [comment]
@edwardloveall Leave this open, we might include it anyway.

**Comment by edwardloveall** [comment]
Sure thing. I can also squash the commits if there's nothing else to change before it's merged. Let me know.

File: Formula/eject.rb [hunk 3]
```diff
@@ -14,6 +14,36 @@ def install
...
+    swift = <<-SWIFT
+// Create Views
+let view = UIView()
+view.frame = CGRect(x: 0, y: 0, width: 375, height: 667)
+view.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+view.backgroundColor = UIColor(red: 1, green: 1, blue: 1, alpha: 1)
+    SWIFT
+
+    system "eject --file view.xib > view.swift"
+    File.read("view.swift") == swift
```
Comment (bfontaine) [inline Formula/eject.rb]: Use `assert_equal swift, shell_output("#{bin}/eject --file view.xib")` instead.

Comment (edwardloveall) [inline Formula/eject.rb]: Done

File: Formula/eject.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  def install
+    xcodebuild
+    bin.install "build/Release/eject.app/Contents/MacOS/eject"
+    frameworks_path = "build/Release/eject.app/Contents/Frameworks"
+    mv frameworks_path, frameworks
+  end
+
+  test do
+    xib = File.open "view.xib", "a+" do |file|
+      file.write <<-XML
```
Comment (MikeMcQuaid) [inline Formula/eject.rb]: Use `EOS.undent` here and then you can indent the line below (and all others) two characters more than before `file.write`

File: Formula/eject.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            <rect key="frame" x="0.0" y="0.0" width="375" height="667"/>
+            <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+            <color key="backgroundColor" red="1" green="1" blue="1" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
+        </view>
+    </objects>
+</document>
+      XML
+    end
+
+    swift = <<-SWIFT
```
Comment (MikeMcQuaid) [inline Formula/eject.rb]: Similarly use `EOS.undent` here.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/eject.rb [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+  def install
+    xcodebuild
+    bin.install "build/Release/eject.app/Contents/MacOS/eject"
+    frameworks_path = "build/Release/eject.app/Contents/Frameworks"
+    mv frameworks_path, frameworks
+  end
+
+  test do
+    xib = File.open "view.xib", "a+" do |file|
```
Comment (MikeMcQuaid) [inline Formula/eject.rb]: Don't need the `xib` assignment here and use `(testpath/"view.xib").write` instead of the block

File: Formula/eject.rb [hunk 7]
```diff
@@ -0,0 +1,48 @@
+class Eject < Formula
+  desc "Generate swift code from Interface Builder xibs"
+  homepage "https://github.com/Raizlabs/Eject"
+  url "https://github.com/Raizlabs/Eject/archive/0.1.12.tar.gz"
+  sha256 "a4dae3d37f780d274f53ed25d9dc1a27d5245289f9b8cbaaf8be71bc9334de18"
+
+  depends_on :xcode => :build
```
Comment (MikeMcQuaid) [inline Formula/eject.rb]: Failing on Yosemite so may need scoped to e.g. Xcode 8: https://bot.brew.sh/job/Homebrew%20Core/11454/version=yosemite/testReport/junit/brew-test-bot/yosemite/install_eject/

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/eject.rb (hunk 6 above)
Comment (edwardloveall) [inline Formula/eject.rb]: 👍

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**